### ISSUE TITLE
frequent exceptions after downloading stationxml

### ISSUE THREAD

**Issue by lileipku00** [obspy/obspy#1304]
I met this exception frequently.

```
[2016-03-01 12:49:16,682] - obspy.clients.fdsn.mass_downloader - INFO: Client 'GFZ' - Downloaded 8 station files [0.2 MB] in 5.6 seconds [31.96 KB/sec].
Traceback (most recent call last):
  File "script_massdownloader.py", line 85, in <module>    
  File "/home/user/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/mass_downloader/mass_downloader.py", line 220, in download
    existing_client_dl_helpers=existing_client_dl_helpers)
  File "/home/user/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/mass_downloader/download_helpers.py", line 565, in filter_stations_based_on_minimum_distance
    existing_stations)
  File "/home/user/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/mass_downloader/utils.py", line 269, in __init__
    self.kd_tree = cKDTree(data=cart_data, leafsize=10)
  File "scipy/spatial/ckdtree.pyx", line 495, in scipy.spatial.ckdtree.cKDTree.__init__ (scipy/spatial/ckdtree.cxx:5441)
  File "/home/user/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py", line 2263, in amax
    out=out, keepdims=keepdims)
  File "/home/user/anaconda/lib/python2.7/site-packages/numpy/core/_methods.py", line 26, in _amax
    return umr_maximum(a, axis, None, out, keepdims)
ValueError: zero-size array to reduction operation maximum which has no identity
./run.sh: error reading input file: Stale NFS file handle
```

**Comment by krischer** [comment]
Hmm...are you running either Python or the mass downloader from an unstable NFS file system?

**Comment by krischer** [comment]
Can you also post the full script here so I can try it on my machine?

**Comment by lileipku00** [comment]
```
import os
from obspy import UTCDateTime
from obspy.clients.fdsn.mass_downloader import RectangularDomain, Restrictions, MassDownloader

providers = ['GFZ']
nthreads = 6
starttime = UTCDateTime('2008-01-01')
endtime = UTCDateTime('2008-12-31')
mseedfold = './MSeed'
xmlfold = './StationXML'

def mseed_filename(network, station, location, channel, starttime, endtime):
    ''' make mseed filename '''
    t = starttime + 0.5*(endtime-starttime)
    year, jday = str(t.year), '%03d'%(t.julday)
    fnm = '.'.join((network, station, location, channel, year, jday))
    return os.path.join(mseedfold, channel[-1], year, jday, fnm)

def xml_filename(network, station, channel, starttime, endtime):
    ''' make mseed filename '''
    return os.path.join(xmlfold, network, '.'.join((station,'xml')))

# domain
domain = RectangularDomain(minlatitude=28, maxlatitude=72, minlongitude=-12, maxlongitude=48)
# restrictions on date and channel
restrictions = Restrictions(starttime=starttime, endtime=endtime,
    sanitize=False, chunklength_in_sec=86400, channel_priorities=('BHZ','HHZ'),  reject_channels_with_gaps=False, minimum_length=0.0, minimum_interstation_distance_in_m=100.0)    
# massive dowload
mdl = MassDownloader(providers=providers)
mdl.download(domain, restrictions, mseed_storage=mseed_filename, stationxml_storage=xml_filename,
             download_chunk_size_in_mb=30, threads_per_client=nthreads, print_report=True)
```

**Comment by lileipku00** [comment]
Another example for LMU

```
[2016-03-01 06:24:50,225] - obspy.clients.fdsn.mass_downloader - INFO: Client 'LMU' - Successfully downloaded '/scratch/user/StationXML/GE/ISP.xml'.
[2016-03-01 06:24:50,225] - obspy.clients.fdsn.mass_downloader - INFO: Client 'LMU' - Successfully downloaded '/scratch/user/StationXML/HL/ITM.xml'.
[2016-03-01 06:24:50,497] - obspy.clients.fdsn.mass_downloader - INFO: Client 'LMU' - Downloaded 92 station files [3.5 MB] in 78.3 seconds [45.75 KB/sec].
Traceback (most recent call last):
  File "script_massdownloader.py", line 85, in <module>
    
  File "/home/user/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/mass_downloader/mass_downloader.py", line 220, in download
    existing_client_dl_helpers=existing_client_dl_helpers)
  File "/home/user/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/mass_downloader/download_helpers.py", line 565, in filter_stations_based_on_minimum_distance
    existing_stations)
  File "/home/user/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/mass_downloader/utils.py", line 269, in __init__
    self.kd_tree = cKDTree(data=cart_data, leafsize=10)
  File "scipy/spatial/ckdtree.pyx", line 495, in scipy.spatial.ckdtree.cKDTree.__init__ (scipy/spatial/ckdtree.cxx:5441)
  File "/home/user/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py", line 2263, in amax
    out=out, keepdims=keepdims)
  File "/home/user/anaconda/lib/python2.7/site-packages/numpy/core/_methods.py", line 26, in _amax
    return umr_maximum(a, axis, None, out, keepdims)
ValueError: zero-size array to reduction operation maximum which has no identity
./run.sh: error reading input file: Stale NFS file handle
```
The error always occurs after the stationxml downloaded. I guess it's no because of the filesystem of the mounted /scratch.

**Comment by lileipku00** [comment]
Another terrible exception is the unknown http code

```
[2016-02-29 23:48:32,578] - obspy.clients.fdsn.mass_downloader - ERROR: Client 'NIEP' - Unknown HTTP code: 502
Detailed response of server:

<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html><head>
<meta type="copyright" content="Copyright (C) 1996-2016 The Squid Software Foundation and contributors">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<title>ERROR: The requested URL could not be retrieved</title>
<style type="text/css"><!-- 
 /*
 Stylesheet for Squid Error pages
 Adapted from design by Free CSS Templates
 http://www.freecsstemplates.org
 Released for free under a Creative Commons Attribution 2.5 License
*/
/* Page basics */
* {
	font-family: verdana, sans-serif;
}
html body {
	margin: 0;
	padding: 0;
	background: #efefef;
	font-size: 12px;
	color: #1e1e1e;
}
/* Page displayed title area */
#titles {
	margin-left: 15px;
	padding: 10px;
	padding-left: 100px;
	background: url('http://www.squid-cache.org/Artwork/SN.png') no-repeat left;
}
/* initial title */
#titles h1 {
	color: #000000;
}
#titles h2 {
	color: #000000;
}
/* special event: FTP success page titles */
#titles ftpsuccess {
	background-color:#00ff00;
	width:100%;
}
/* Page displayed body content area */
#content {
	padding: 10px;
	background: #ffffff;
}
/* General text */
p {
}
/* error brief description */
#error p {
}
/* some data which may have caused the problem */
#data {
}
/* the error message received from the system or other software */
#sysmsg {
}
pre {
    font-family:sans-serif;
}
/* special event: FTP / Gopher directory listing */
#dirmsg {
    font-family: courier;
    color: black;
    font-size: 10pt;
}
#dirlisting {
    margin-left: 2%;
    margin-right: 2%;
}
#dirlisting tr.entry td.icon,td.filename,td.size,td.date {
    border-bottom: groove;
}
#dirlisting td.size {
    width: 50px;
    text-align: right;
    padding-right: 5px;
}
/* horizontal lines */
hr {
	margin: 0;
}
/* page displayed footer area */
#footer {
	font-size: 9px;
	padding-left: 10px;
}
body
:lang(fa) { direction: rtl; font-size: 100%; font-family: Tahoma, Roya, sans-serif; float: right; }
:lang(he) { direction: rtl; }
 --></style>
</head><body id=ERR_READ_ERROR>
<div id="titles">
<h1>ERROR</h1>
<h2>The requested URL could not be retrieved</h2>
</div>
<hr>
<div id="content">
<p>The following error was encountered while trying to retrieve the URL: <a href="http://eida-sc3.infp.ro/fdsnws/dataselect/1/query">http://eida-sc3.infp.ro/fdsnws/dataselect/1/query</a></p>
<blockquote id="error">
<p><b>Read Error</b></p>
</blockquote>
<p id="sysmsg">The system returned: <i>(54) Connection reset by peer</i></p>
<p>An error condition occurred while reading data from the network. Please retry your request.</p>
<p>Your cache administrator is <a href="mailto:dsi-systeme@ujf-grenoble.fr?subject=CacheErrorInfo%20-%20ERR_READ_ERROR&amp;body=CacheHost%3A%20www-cache-2%0D%0AErrPage%3A%20ERR_READ_ERROR%0D%0AErr%3A%20(54)%20Connection%20reset%20by%20peer%0D%0ATimeStamp%3A%20Mon,%2029%20Feb%202016%2022%3A44%3A58%20GMT%0D%0A%0D%0AClientIP%3A%20195.220.82.35%0D%0AServerIP%3A%20eida-sc3.infp.ro%0D%0A%0D%0AHTTP%20Request%3A%0D%0APOST%20%2Ffdsnws%2Fdataselect%2F1%2Fquery%20HTTP%2F1.1%0AContent-Length%3A%2068%0D%0AAccept-Encoding%3A%20gzip%0D%0AConnection%3A%20close%0D%0AUser-Agent%3A%20ObsPy%201.0.0%20(Linux-3.2.0-4-amd64-x86_64-with-debian-7.8,%20Python%202.7.11)%0D%0AHost%3A%20eida-sc3.infp.ro%0D%0AContent-Type%3A%20application%2Fx-www-form-urlencoded%0D%0A%0D%0A%0D%0A">dsi-systeme@ujf-grenoble.fr</a>.</p>
<br>
</div>
<hr>
<div id="footer">
<p>Generated Mon, 29 Feb 2016 22:44:58 GMT by www-cache-2 (squid/3.5.14)</p>
<!-- ERR_READ_ERROR -->
</div>
</body></html>
```

This can give rise to a log file beyond 500M. A lot of html codes were written into the log file.

**Comment by lileipku00** [comment]
And for RESIF, it seems it doesn't support the parameter `matchtimeseries`. Changing itl to false makes the program work.
```[2016-02-29 23:34:38,849] - obspy.clients.fdsn.mass_downloader - ERROR: Client 'RESIF' - Failed getting availability: Bad request. If you think your request was valid please contact the developers.
Detailed response of server:

Error 400: handlerProgram exit code: 3  handlerProgram message: Parameter not implemented, please query with default value instead : matchtimeseries=FALSE
Some of your query parameters are invalid or non-coherent. Please mention your request URL when asking for support.
Request:
http://ws.resif.fr/fdsnws/station/1/query?endtime=2011-12-31T00%3A00%3A00.000000&maxlongitude=48.0&format=text&level=channel&minlongitude=-12.0&maxlatitude=72.0&starttime=2011-01-01T00%3A00%3A00.000000&matchtimeseries=true&minlatitude=28.0
```

BTW, I cannot understand the document about `matchtimeseries` very well. What happens on the fdsn server if it is true? Can somebody give me some explanation?

**Comment by krischer** [comment]
Cool. Thanks for the thorough reports! Its always very helpful to get other opinions and experiences.

#### Stale NFS file handle

`./run.sh: error reading input file: Stale NFS file handle`: The script downloads a humongous amount of data but works just fine on my machine. I suspect it is an issue with your installation - it seems to happen when the kd-tree is initialized. Can you modify `/home/user/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/mass_downloader/utils.py` and print `cart_data` before the line containing `self.kd_tree = cKDTree(data=cart_data, leafsize=10)`? This might enable you to generate a minimal code snippet that reproduces the error. Also: Can you run the `scipy` tests? Another avenue to debug the issue is to ask your network administrator the check the NFS logs for any strange behavior.

```bash
$ python -c "import scipy; scipy.test()"
```

Trying to update scipy might also be worthwhile.

#### HTTP Code 502

This is not defined by the fdns-ws and I think the mass downloader handles it in the best possible way. If an unexpected error occurs we do want to log it. We should pass this to the NIEP for two reasons: (1) this should not happen, and (2) its better to return raw text error messages for services that are largely intended to be queried by code. Can you run once again with `MassDownloader(providers=["NEIP"], debug=True)` which prints a lot of information but should enable you to find the problematic queries.


#### matchtimeseries

AFAIK the IRIS DMC is the only service implementing the `matchtimeseries` parameter. Its a non-standard "hack" that solves a glaring issue with the current FDSN web services: It enables availability queries based on the actually available data. In short: If `matchtimeseries` is true, the service should only return station information for stations where it actually has data in the requested time span. This enables the mass downloader to select what to download before it actually downloads things - otherwise it has to see what it can actually get before deciding what to keep.

In this case the `application.wadl` file of RESIF is just wrong as it claims to support the `matchtimeseries` parameter but it actually does not. I don't think we should special case this from our side as it is the responsibility of the service providers to maintain accurate `application.wadl` files. I contacted RESIF about this issue and I hope they resolve it soon. The mass downloader should work with RESIF once they fix it.

http://ws.resif.fr/fdsnws/station/1/application.wadl

**Comment by lileipku00** [comment]
`python -c "import scipy; scipy.test()"`
generates
```
Running unit tests for scipy
NumPy version 1.10.4
NumPy relaxed strides checking option: False
NumPy is installed in /home/user/anaconda/lib/python2.7/site-packages/numpy
SciPy version 0.17.0
SciPy is installed in /home/user/anaconda/lib/python2.7/site-packages/scipy
Python version 2.7.11 |Anaconda 2.3.0 (64-bit)| (default, Dec  6 2015, 18:08:32) [GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
nose version 1.3.7
/home/user/anaconda/lib/python2.7/site-packages/numpy/lib/utils.py:99: DeprecationWarning: `scipy.weave` is deprecated, use `weave` instead!
  warnings.warn(depdoc, DeprecationWarning)
.....................................................................................................................................................................................................................................................................K..................................................................................................................................................................................................................................................................................................................K.............................................................................................................................................................K..K................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................0-th dimension must be fixed to 3 but got 15
..0-th dimension must be fixed to 3 but got 5
..............0-th dimension must be fixed to 4 but got 2
0-th dimension must be fixed to 4 but got 2
.0-th dimension must be fixed to 4 but got 2
0-th dimension must be fixed to 4 but got 2
.0-th dimension must be fixed to 4 but got 2
0-th dimension must be fixed to 4 but got 2
0-th dimension must be fixed to 4 but got 2
0-th dimension must be fixed to 4 but got 2
0-th dimension must be fixed to 4 but got 2
0-th dimension must be fixed to 4 but got 2
................................K...........................................................................................................................................................................................................................................................................................................K..................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................Segmentation fault

```

**Comment by lileipku00** [comment]
Besides 502, there is another unknown http code 404 from SCEDC (as well as unsupported parameters).

```
[2016-02-29 23:34:38,899] - obspy.clients.fdsn.mass_downloader - INFO: Client 'SCEDC' - Requesting unreliable availability.
/home/user/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/client.py:1141: UserWarning: The standard parameter 'maxlongitude' is not supported by the webservice. It will be silently ignored.
  warnings.warn(msg)
/home/user/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/client.py:1141: UserWarning: The standard parameter 'minlatitude' is not supported by the webservice. It will be silently ignored.
  warnings.warn(msg)
/home/user/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/client.py:1141: UserWarning: The standard parameter 'minlongitude' is not supported by the webservice. It will be silently ignored.
  warnings.warn(msg)
/home/user/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/client.py:1141: UserWarning: The standard parameter 'maxlatitude' is not supported by the webservice. It will be silently ignored.
  warnings.warn(msg)
[2016-02-29 23:35:02,899] - obspy.clients.fdsn.mass_downloader - INFO: Client 'SCEDC' - Successfully requested availability (24.00 seconds)
[2016-03-01 00:01:18,442] - obspy.clients.fdsn.mass_downloader - INFO: Client 'SCEDC' - Found 33 stations (36 channels).
[2016-03-01 00:01:18,545] - obspy.clients.fdsn.mass_downloader - INFO: Client 'SCEDC' - Will attempt to download data from 33 stations.
[2016-03-01 00:01:39,865] - obspy.clients.fdsn.mass_downloader - INFO: Client 'SCEDC' - Status for 592 time intervals/channels before downloading: EXISTS
[2016-03-01 00:01:39,865] - obspy.clients.fdsn.mass_downloader - INFO: Client 'SCEDC' - Status for 12512 time intervals/channels before downloading: NEEDS_DOWNLOADING
[2016-03-01 00:01:41,009] - obspy.clients.fdsn.mass_downloader - ERROR: Client 'SCEDC' - Service responds: Internal server error
Detailed response of server:

Error 500: handlerProgram exit code: 1  handlerProgram message: Traceback (most recent call last):
  File "/www/wssconfig/dataselect-handler.py", line 423, in <module>
    Main()
  File "/www/wssconfig/dataselect-handler.py", line 39, in Main
    retval = HandleRequest(params, logfp)    
  File "/www/wssconfig/dataselect-handler.py", line 167, in HandleRequest
    logfp.write(etype + "\n")
TypeError: unsupported operand type(s) for +: 'type' and 'str'
Request:
http://service.scedc.caltech.edu/fdsnws/dataselect/1/query
Request Submitted:
2016/02/29 22:58:06 UTC
Service version:
fdsnws-dataselect: v 1.1.0
[2016-03-01 00:01:42,402] - obspy.clients.fdsn.mass_downloader - ERROR: Client 'SCEDC' - Unknown HTTP code: 404
Detailed response of server:

Error 404: handlerProgram exit code: 2  handlerProgram message: No data found for search criteria
Request:
http://service.scedc.caltech.edu/fdsnws/dataselect/1/query
Request Submitted:
2016/02/29 22:58:08 UTC
Service version:
fdsnws-dataselect: v 1.1.0
```

**Comment by megies** [comment]
Any chance that the installed scipy package in conda is compiled against a different numpy than the installed one?

Can you do a `conda list | grep -e "np" -e "numpy"`?

**Comment by krischer** [comment]
Yes - your whole installation appears to be really messed up, see also #1294. Maybe its time to take it to your admin as there appears to be some severe misconfiguration of some parts of the system.

**Comment by krischer** [comment]
The SCEC service has a couple of issues and all the things you noted should be fixed from their side. I sent them a message and hope they'll fix them soon.

**Comment by megies** [comment]
Btw.. @krischer last week at the workshop when internet was down (wlan connected but router not granting access to internet) i ran into another uncaught http code for the fdsn client, resulting in a server response getting fed to MSEED reader routine.. didn't have time to document properly at the time and can't reproduce right now though..

**Comment by krischer** [comment]
Yea - it downloaded the login page of the hotel internet which every HTTP request was redirected to it one was kicked out of the net...that is really nothing we can catch from our side - as soon as the internet connection is being tampered with it might just break.

**Comment by megies** [comment]
Oh, yeah, right. Nothing we can do about that one.. correct.

**Comment by lileipku00** [comment]
@megies 

`$ conda list | grep -e "np" -e "numpy"`

on my PC
```
abstract-rendering        0.5.1                np19py27_0    <unknown>
anaconda                  2.3.0                np19py27_0    defaults
astropy                   1.0.3                np19py27_0    defaults
basemap                   1.0.7                np19py27_0    defaults
bcolz                     0.9.0                np19py27_0    defaults
blaze-core                0.8.0                np19py27_0    defaults
blz                       0.6.2                np19py27_1    defaults
bokeh                     0.9.0                np19py27_0    defaults
bottleneck                1.0.0                np19py27_0    defaults
datashape                 0.4.5                np19py27_0    defaults
dynd-python               0.6.5                np19py27_0    <unknown>
h5py                      2.5.0               np110py27_4    defaults
matplotlib                1.5.1               np110py27_0    defaults
mayavi                    4.4.0                np19py27_0    defaults
netcdf4                   1.2.2               np110py27_0    defaults
nltk                      3.0.3                np19py27_0    defaults
numba                     0.19.1               np19py27_0    defaults
numexpr                   2.4.6               np110py27_1    defaults
numpy                     1.10.4                   py27_0    defaults
odo                       0.3.2                np19py27_0    defaults
openpyxl                  1.8.5                    py27_0    <unknown>
pandas                    0.17.0              np110py27_0    defaults
patsy                     0.3.0                np19py27_0    <unknown>
pytables                  3.2.2               np110py27_0    defaults
scikit-image              0.11.3               np19py27_0    defaults
scikit-learn              0.16.1               np19py27_0    defaults
scipy                     0.17.0              np110py27_1    defaults
statsmodels               0.6.1                np19py27_0    <unknown>
theano                    0.7.0                np19py27_0    defaults
```
on one server
```
abstract-rendering        0.5.1                np19py27_0    <unknown>
anaconda                  2.0.0                np18py27_0    defaults
astropy                   0.3.2                np18py27_0    defaults
basemap                   1.0.7               np110py27_0    defaults
bcolz                     0.8.1                np19py27_0    <unknown>
blaze                     0.5.0                np18py27_0    defaults
blaze-core                0.7.3                np19py27_0    <unknown>
blz                       0.6.2                np18py27_0    defaults
bokeh                     0.4.4                np18py27_1    defaults
chaco                     4.4.1                np18py27_0    defaults
datashape                 0.2.0                np18py27_1    defaults
dynd-python               0.6.2                np18py27_0    defaults
enable                    4.3.0                np18py27_2    defaults
h5py                      2.3.0                np18py27_0    defaults
matplotlib                1.5.1               np110py27_0    defaults
nltk                      2.0.4                np18py27_0    defaults
numba                     0.13.1               np18py27_0    defaults
numexpr                   2.3.1                np18py27_0    defaults
numpy                     1.10.4                   py27_0    defaults
odo                       0.3.1                np19py27_0    <unknown>
openpyxl                  2.0.2                    py27_0    defaults
pandas                    0.13.1               np18py27_0    defaults
patsy                     0.2.1                np18py27_0    defaults
pytables                  3.1.1                np18py27_0    defaults
scikit-image              0.9.3                np18py27_0    defaults
scikit-learn              0.14.1               np18py27_1    defaults
scipy                     0.17.0              np110py27_1    defaults
statsmodels               0.5.0                np18py27_0    defaults
theano                    0.6.0                np18py27_0    defaults
```
on another server
```
abstract-rendering        0.5.1                np19py27_0    <unknown>
anaconda                  2.3.0                np19py27_0    <unknown>
astropy                   1.0.3                np19py27_0    <unknown>
bcolz                     0.9.0                np19py27_0    <unknown>
blaze-core                0.8.0                np19py27_0    <unknown>
blz                       0.6.2                np19py27_1    <unknown>
bokeh                     0.9.0                np19py27_0    <unknown>
bottleneck                1.0.0                np19py27_0    <unknown>
datashape                 0.4.5                np19py27_0    <unknown>
dynd-python               0.6.5                np19py27_0    <unknown>
h5py                      2.5.0                np19py27_3    <unknown>
matplotlib                1.5.1               np110py27_0    defaults
nltk                      3.0.3                np19py27_0    <unknown>
numba                     0.19.1               np19py27_0    <unknown>
numexpr                   2.4.6               np110py27_1    defaults
numpy                     1.10.4                   py27_0    defaults
odo                       0.3.2                np19py27_0    <unknown>
openpyxl                  1.8.5                    py27_0    <unknown>
pandas                    0.16.2               np19py27_0    <unknown>
patsy                     0.3.0                np19py27_0    <unknown>
pytables                  3.2.2               np110py27_0    defaults
scikit-image              0.11.3               np19py27_0    <unknown>
scikit-learn              0.16.1               np19py27_0    <unknown>
scipy                     0.17.0              np110py27_1    defaults
statsmodels               0.6.1                np19py27_0    <unknown>
theano                    0.7.0                np19py27_0    <unknown>
```

**Comment by lileipku00** [comment]
@krischer 

I installed in this way.
1. install `Anaconda` in `~/anaconda` and update `$PATH`
2. `conda install -c obspy obspy` (it updates lots of modules)
3. possibly `conda update pip`, `conda update spyder`, ... (lots of modules updated meanwhile)

Since I ran codes by anaconda python, theoretically there is nothing to do with the system things.

**Comment by krischer** [comment]
True - but if the system configuration interferes it might mix things and all bets are off. It might be worthwhile to check the environment variables

```bash
$ env
```

Also try to use conda environments which offer a cleaner separation from the system python.

**Comment by lileipku00** [comment]
@krischer I inserted 
`print('cart_data type: '+type(cart_data))`
and 
`print('cart_data: '+str(cart_data))`
and ran `debug=True`. It shows that `cart_data` is empty and there is no mseed saved to disk.

```
Downloaded http://www.orfeus-eu.org/fdsnws/dataselect/1/query with HTTP code: 204
Downloaded http://www.orfeus-eu.org/fdsnws/dataselect/1/query with HTTP code: 204
cart_data type: <type 'numpy.ndarray'>
cart_data: []

Traceback (most recent call last):
  File "script_massdownloader_lastdoy.py", line 65, in <module>
    download_chunk_size_in_mb=30, threads_per_client=nthreads, print_report=True)
  File "/home/lile/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/mass_downloader/mass_downloader.py", line 220, in download
    existing_client_dl_helpers=existing_client_dl_helpers)
  File "/home/lile/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/mass_downloader/download_helpers.py", line 537, in filter_stations_based_on_minimum_distance
    kd_tree = utils.SphericalNearestNeighbour(stations)
  File "/home/lile/anaconda/lib/python2.7/site-packages/obspy/clients/fdsn/mass_downloader/utils.py", line 273, in __init__
    self.kd_tree = cKDTree(data=cart_data, leafsize=10)
  File "scipy/spatial/ckdtree.pyx", line 495, in scipy.spatial.ckdtree.cKDTree.__init__ (scipy/spatial/ckdtree.cxx:5441)
  File "/home/lile/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py", line 2263, in amax
    out=out, keepdims=keepdims)
  File "/home/lile/anaconda/lib/python2.7/site-packages/numpy/core/_methods.py", line 26, in _amax
    return umr_maximum(a, axis, None, out, keepdims)
ValueError: zero-size array to reduction operation maximum which has no identity

```

**Comment by lileipku00** [comment]
So it is caused when no data is available on the server?

**Comment by krischer** [comment]
I could finally reproduce your issue with the kd-tree. It was an unlucky combination of requests and provider order...should be fixed with c942e6df9660b6300735955ca33426a109ecd4de in the maintenance branch. Can you check?

> Besides, there is another exception UnicodeDecodeError raised during the debug but not terminating the program.

Can you initialize with `MassDownloader(..., debug=True)`? This should print the URL where it actually goes wrong.

**Comment by krischer** [comment]
Should be fixed. Please reopen if it still does not work.
